caas/kubernetes/provider/image.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package provider
+
+import (
+	"crypto/rand"
+	"fmt"
+	"io"
+
+	"github.com/juju/errors"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Constants below are copied from "k8s.io/kubernetes/pkg/kubelet/images"
+// to avoid introducing the huge dependency.
+var (
+	// errImagePullBackOff - Container image pull failed, kubelet is backing off image pull
+	errImagePullBackOff = "ImagePullBackOff"
+	// errImageInspect - Unable to inspect image
+	errImageInspect = "ImageInspectError"
+	// errImagePull - General image pull error
+	errImagePull = "ErrImagePull"
+	// errImageNeverPull - Required Image is absent on host and PullPolicy is NeverPullImage
+	errImageNeverPull = "ErrImageNeverPull"
+	// errRegistryUnavailable - Get http error when pulling image from registry
+	errRegistryUnavailable = "RegistryUnavailable"
+	// errInvalidImageName - Unable to parse the image name.
+	errInvalidImageName = "InvalidImageName"
+)
+
+func (k *kubernetesClient) operatorImagePrepullCheck(image string) error {
```

Comment:
This seems somewhat heavy?
We already have juju/juju/docker/docker.go which has the method ListOperatorImages()
We use this when upgrading a controller to see what the available images are. We can use that here too. There's no issue with secrets/access because all juju operator images are public.